### ISSUE TITLE
`add_real_directory` does not work after chdir

### ISSUE THREAD

**Issue by abbec** [jmcgeheeiv/pyfakefs#355]
If I do an `os.chdir()` call before calling `add_real_directory` with an absolute path, `os.listdir()` on that directory returns nothing since target path is set to the combination of the current directory and the specified path to `add_real_directory`.

Example

```python
os.chdir("/Users/abbe/test")
fs.add_real_directory("/Users/abbe/real")
os.listdir("/Users/abbe/real")
```

the call to `os.listdir()` returns nothing in this case since the file gets added as `/Users/abbe/test/Users/abbe/real` instead of `/Users/abbe/real`.

**Comment by mrbean-bremen** [comment]
Thanks for the report - I will have a look!

**Comment by mrbean-bremen** [comment]
So, to understand this: this is inside a unit test, or after setting up Patcher? E.g. `os` here is the faked version?
In this case, you already have a directory "/Users/abbe/test" in your fake file system - did you add that also using `add_real_directory()`?

**Comment by mrbean-bremen** [comment]
I'm trying to reproduce this, but until now it works correctly for me - so I'm obviously doing something different. I need a bit more context to be able to reproduce this...

**Comment by abbec** [comment]
Yes, this is inside a unit test/after patcher is set up (`os` is the faked version) and `/Users/abbe/test` was added to the fake filesystem with `create_dir()` prior to the above lines.

**Comment by abbec** [comment]
This code

```python
import os

def test_something(fs):
    fs.create_dir("/Users/acervin/somethingfake")
    os.chdir("/Users/acervin/somethingfake")
    fs.add_real_directory("/Users/acervin/test")

    print(os.listdir("/Users/acervin/test"))
    print(os.listdir("/Users/acervin/somethingfake/Users/acervin/test"))

    assert os.listdir("/Users/acervin/test")
```

gives this output

![screen shot 2018-03-13 at 09 34 57](https://user-images.githubusercontent.com/455022/37330571-da8a5dc2-26a1-11e8-8e78-1203116baa86.png)

and the folder `/Users/acervin/test` contains one file called `test.txt`.

**Comment by abbec** [comment]
And if I remove the call to `os.chdir` in the above code (effectively leaving cwd at `/`) it works.

**Comment by mrbean-bremen** [comment]
Ok, thanks! I will have another lokk in the evening.

**Comment by mrbean-bremen** [comment]
Unfortunately I couldn't reproduce this under Windows, so I think it is Unix-specific. My Ubuntu got trashed with the latest Windows 10 reinstall, and I couldn't get the grub menu to show up even after an Ubuntu reinstall, grub-update and the like, so I'm giving up for today. Got to love Windows 10...
I may have another go at the weekend.

**Comment by abbec** [comment]
Yeah, it is specific to *NIX and I think it caused by the code here: https://github.com/jmcgeheeiv/pyfakefs/blob/master/pyfakefs/fake_filesystem.py#L394. When it is normalizing it ignores the leading slash (that denotes the root) since it says `names[1:]`. The function `absnormpath` will then hit this code:

```python
elif not self._starts_with_root_path(path):
    # Prefix relative paths with cwd, if cwd is not root.
    ...
```

**Comment by mrbean-bremen** [comment]
Thanks - that helped! So the real bug is in `FakeDirectory.path`, I just added a fix for this. Please check if this fixes your problem.

**Comment by abbec** [comment]
Awesome thanks!
